rs/types/types/src/messages.rs

```diff
@@ -520,21 +520,22 @@ impl Display for CanisterMessageOrTask {
 /// A wrapper around canister messages and tasks.
 #[derive(Clone, Eq, PartialEq, Hash, Debug)]
 pub enum CanisterCallOrTask {
-    Call(CanisterCall),
+    Update(CanisterCall),
+    Query(CanisterCall),
     Task(CanisterTask),
 }
```

Comment:
It may be worth mentioning somewhere in here that these are replicated executions (i.e. `Query` never stands for a non-replicated query.

Reply:
I can enhance the doc comment. The name of the enum (although convoluted) should capture it. `call` is usually meant to be something executed in replicated mode.